server.js

```diff
@@ -1,23 +1,88 @@
 const express = require("express");
 const cors = require("cors");
+const bodyParser = require("body-parser");
+
+const PORT = process.env.PORT || 3000;
 
 const app = express();
+app.use(bodyParser.urlencoded({ extended: true }));
 
 app.use(cors());
 
 const welcomeMessage = {
   id: 0,
-  from: "Bart",
+  from: "Joanna",
   text: "Welcome to CYF chat system!",
+  timeSent: 28 / 12 / 2022,
 };
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
 
-app.get("/", function (request, response) {
-  response.sendFile(__dirname + "/index.html");
+app.get("/messages/search", (req, res) => {
+  const { term } = req.query;
+  console.log(term);
+
+  const filterMessages = messages.filter((message) =>
+    message.text.toLocaleLowerCase().includes(term.toLocaleLowerCase())
+  );
+  console.log(filterMessages);
+  res.send(filterMessages);
+});
+
+app.get("/", (req, res) => {
+  res.sendFile(__dirname + "/index.html");
+});
+
+//All messages 
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+
+//Last 10 messages
+
+app.get("/messages/latest", (req, res) => {
+  res.send(messages.slice(-10))
+})
+
+//New message
+app.post("/messages", (req, res) => {
+  const { from, text } = req.body;
+
+  const messageObject = {
+    id: messages.length + 1,
```

Comment:
this method of selecting an id can lead to duplicate id values.

Imaging you had a set of 20 messages, where the last message is id:20. You delete message 19. When you next send a message, it will also be given id:20